src/main/java/com/monumental/services/MonumentService.java

```diff
@@ -44,40 +44,94 @@ private void buildSimilarityQuery(CriteriaBuilder builder, CriteriaQuery query,
+     * May make use of the pg_trgm similarity or ST_DWithin functions
+     * @param searchQuery - The string search query that will get passed into the pg_tgrm similarity function
+     * @param page - The page number of Monument results to return
+     * @param limit - The maximum number of Monument results to return
+     * @param latitude - The latitude of the comparison point
+     * @param longitude - The longitude of the comparison point
+     * @param miles - The number of miles from the comparison point to search in
+     * @return List<Monument> - List of Monument results based on the specified search parameters
      */
     @SuppressWarnings("unchecked")
-    public List<Monument> search(String searchQuery, String page, String limit) {
+    public List<Monument> search(String searchQuery, String page, String limit, String latitude, String longitude,
+                                 String miles) {
 
         CriteriaBuilder builder = this.getCriteriaBuilder();
         CriteriaQuery<Monument> query = this.createCriteriaQuery(builder, false);
         Root<Monument> root = this.createRoot(query);
         root.fetch("tags", JoinType.LEFT);
         query.select(root);
 
+        // TODO: Query for description
         // TODO: Query for tags
-        // TODO: Query for location
         // TODO: Filters
 
         if (searchQuery != null) {
             this.buildSimilarityQuery(builder, query, root, searchQuery, 0.1, true);
         }
 
-        return this.getWithCriteriaQuery(query, Integer.parseInt(limit), (Integer.parseInt(page)) - 1);
+        if (latitude != null && longitude != null && miles != null) {
+            this.buildDWithinQuery(builder, query, root, latitude, longitude, Integer.parseInt(miles));
+        }
+
+        List<Monument> monuments = this.getWithCriteriaQuery(query, Integer.parseInt(limit), (Integer.parseInt(page)) - 1);
+
+        return monuments;
     }
 
     /**
-     * Count the total number of results for a FTS search
+     * Count the total number of results for a Monument search
      * TODO: Be sure to update this whenever the search function changes so that they stay in sync
      */
-    public Integer countSearchResults(String searchQuery) {
+    public Integer countSearchResults(String searchQuery, String latitude, String longitude, String miles) {
 
         CriteriaBuilder builder = this.getCriteriaBuilder();
         CriteriaQuery<Long> query = builder.createQuery(Long.class);
         Root<Monument> root = query.from(Monument.class);
         query.select(builder.countDistinct(root));
 
-        this.buildSimilarityQuery(builder, query, root, searchQuery, 0.1, false);
+        if (searchQuery != null) {
+            this.buildSimilarityQuery(builder, query, root, searchQuery, 0.1, false);
+        }
+
+        if (latitude != null && longitude != null && miles != null) {
+            this.buildDWithinQuery(builder, query, root, latitude, longitude, Integer.parseInt(miles));
+        }
```

Comment:
I'm realizing now that this whole thing will always be a duplicate of what's in `search` so we should really abstract it to a function. Can be in a different PR though, like when we add filters

Reply:
Agreed.